src/move_ref.rs

```diff
@@ -189,27 +207,59 @@ impl<'a, T> From<MoveRef<'a, T>> for Pin<MoveRef<'a, T>> {
+/// The first part of this consists of converting the pointer into the
+/// "partially deinitialized" form, represented by the type
+/// [`DerefMove::Storage`]: it is the pointer as "pure storage".
 ///
-/// The correct way to perform a `DerefMove` operation is thus:
-/// ```
-/// # use moveit::{DerefMove, MoveRef, Slot, slot};
-/// # slot!(x: i32);
-/// # let p = x.put(5);
-/// # type MyPtr<'a> = MoveRef<'a, i32>;
-/// let mut deinit = MyPtr::deinit(p);
-/// let move_ref = unsafe { MyPtr::deref_move(&mut deinit) };
-/// ```
+/// This pointer should be placed into the [`DroppingSlot`] passed into
+/// `deref_move`, so that it has a fixed lifetime for the duration of the frame
+/// that the [`MoveRef`] will exist for. The [`DroppingSlot`] will also provide
+/// a drop flag to use to build the returned [`MoveRef`].
+///
+/// The mutable reference returned by the [`DroppingSlot`] should then be
+/// converted into a [`MoveRef`]. The end result is that the [`DroppingSlot`]
+/// owns the "outer" part of the pointer, while the [`MoveRef`] owns the "inner"
+/// part. The `'frame` lifetime enforces the correct destruction order of these
+/// two parts, since the [`MoveRef`] borrows the [`DroppingSlot`].
+///
+/// The [`moveit!()`] macro helps by constructing the [`DroppingSlot`] for you.
+///
+/// ## Worked Example: [`Box<T>`]
+///
+/// To inhibit the inner destructor of [`Box<T>`], we can use `Box<MaybeUninit<T>>`
+/// as [`DerefMove::Storage`]. [`MaybeUninit`] is preferred over [`ManuallyDrop`],
+/// since it helps avoid otherwise scary aliasing problems with `Box<&mut T>`.
+///
+/// The first step is to "cast" `Box<T>` into `Box<MaybeUninit<T>>` via
+/// [`Box::into_raw()`] and [`Box::from_raw()`]. This is then placed into the
+/// final storage location using [`DroppingSlot::put()`].
+///
+/// This returns a `&mut Box<MaybeUninit<T>>` and a [`DropFlag`]; the former is
+/// converted into an `&mut T` via [`MaybeUninit::assume_init_mut()`].
 ///
-/// The [`moveit!()`]` macro can do this safely in a single operation.
+/// Finally, [`MoveRef::new_unchecked()`] is used to combine these into the
+/// return value.
+///
+/// The first step is safe because we construct a `MoveRef` to reinstate the
+/// destructor at the end of the function. The second step is safe because
+/// we know, a priori, that the `Box` contains an initialized value. The final
+/// step is safe, because we know, a priori, that the `Box` owns its pointee.
+///
+/// The use of the drop flag in this way makes it so that dropping the resulting
+/// `MoveRef` will leak the storage on the heap, exactly the same way as if we
+/// had leaked a `Box`.
+///
+/// ## Worked Example: [`MoveRef<T>`]
+///
+/// We don't need to inhibit any destructors: we just need to convert a
+/// `MoveRef<MoveRef<T>>` into a `MoveRef<T>`, which we can do by using
+/// [`MoveRef::into_inner()`]. [`DerefMove::Storage`] can be whatever, so we
+/// choose [`()`] for this; the choice is not actually important!
```

Comment:
Why `()` instead of a generic?  If it doesn't matter, why does it get fixed?  Or are there casts everywhere?